### DISCUSSION TITLE
マスクの使い方について

### DISCUSSION THREAD

**Discussion by hadoumune** [kyubuns/AkyuiUnity#34] [Q&A] [answered]
XD上でシェイプマスクを使用しているのですが、Unityにインポートするとマスクが付いていない状態になります。
マスクの適用の仕方が分かるドキュメントがあると助かります。

**Comment by hadoumune** [comment]
自己解決しました
・マスクイメージのFill(塗り）にチェックを入れる
・Unity側でマスクイメージのインポート設定からαを無効にし、compressが効かないフォーマットにする（圧縮フォーマットが自由なサイズを受け付けないため）
・可能であればR8に変換する
とする事でマスクが適用されるようになりました。
Fillにチェックが入っていないとXD上ではマスクが効いてもインポートした際にマスクテクスチャが生成されないようです。
またマスクだけ特殊なテクスチャフォーマットにする必要があるため、なんらかイメージのインポート時のトリガーもしくはマスクだけ特殊なフォーマットに変換してもらえると助かります。

**Comment by kyubuns** [comment, marked as answer]
調査までしていただきありがとうございます。
「MaskImageにAlphaが含まれている画像やシェイプを利用した場合に、XD上とUnity上で描画結果が変わってしまう問題」として、
対応を検討しようと思います。
